uint32 message_type
string data